### PR TITLE
cardList and "addCard func" using redux.

### PR DESCRIPTION
**Pull request by mun625**
implemented simple cards, cardList, and addFunctions using redux.
we will add test code right after this commit.

@fguy @shberry16 @yuda1124

### CODE REVIEW

File: src/shared/components/MapBlock.js [hunk 1]
```diff
@@ -1,21 +1,19 @@
 import React, {PropTypes, Component} from 'react';
+import { connect } from 'react-redux';
 import controllable from 'react-controllables';
 
 import GoogleMap from 'google-map-react';
 import Marker from './Marker.js';
 import CardList from './CardList.js';
-import update from 'react-addons-update';
+
```
Comment (ovekyc) [inline src/shared/components/MapBlock.js]: Too many new line

File: src/shared/actions/map.js [hunk 2]
```diff
@@ -30,4 +30,11 @@ export function getMapMarkers() {
   };
 }
 
-
+let nextCardId = 0;
```
Comment (fguy) [inline src/shared/actions/map.js:34]: What would you like to do with this ID? Do you need sequential ID? Use UUID4 if you just need unique ID: https://www.npmjs.com/package/uuid4

File: src/shared/components/MapBlock.js [hunk 3]
```diff
@@ -71,6 +97,20 @@ MapBlock.defaultProps = {
...
+    mapClick: () => {
+      dispatch(addCard());
+    }
+  };
+};
+
 MapBlock = controllable(MapBlock, ['center', 'zoom', 'markers']);
 
-export default MapBlock;
+export default connect(mapStateToProps, mapDispatchToProps)(MapBlock);
```
Comment (fguy) [inline src/shared/components/MapBlock.js]: connect is a role for container. Please extract out line 100 - 116 as a container.

File: src/shared/components/MapBlock.js [hunk 4]
```diff
@@ -21,6 +25,17 @@ class MapBlock extends Component {
     this.props.onCenterChange([childProps.lat, childProps.lng]);
   }
 
+/*
```
Comment (fguy) [inline src/shared/components/MapBlock.js]: Please do not push the code that commented out: https://medium.com/@kentcdodds/please-don-t-commit-commented-out-code-53d0b5b26d5f

File: src/shared/components/MapBlock.js [hunk 5]
```diff
@@ -1,13 +1,17 @@
 import React, {PropTypes, Component} from 'react';
+import { connect } from 'react-redux';
 import controllable from 'react-controllables';
 
 import GoogleMap from 'google-map-react';
 import Marker from './Marker.js';
+import CardList from './CardList.js';
+
```
Comment (fguy) [inline src/shared/components/MapBlock.js]: Please do not line break between imports.

**Comment by fguy** [comment]
Tests?

**Comment by mun625** [comment]
will add test now.

But before that, I am making cardList containers independent from MapBlock Component.

File: src/shared/actions/map.js [hunk 6]
```diff
@@ -30,4 +30,13 @@ export function getMapMarkers() {
...
-
+/*
+let nextCardId = 0;
+export const addCard = () => {
+  nextCardId = nextCardId + 1;
+  return ({
+    type: 'ADD_CARD',
+    id: nextCardId
+  });
+};
```
Comment (ochanje210) [inline src/shared/actions/map.js:41]: why do we need this comments

File: src/shared/components/MapBlock.js [hunk 7]
```diff
@@ -20,7 +19,7 @@ class MapBlock extends Component {
   onChildClick(key, childProps) {
     this.props.onCenterChange([childProps.lat, childProps.lng]);
   }
-
+
```
Comment (fguy) [inline src/shared/components/MapBlock.js:22]: Trailing space

**Comment by fguy** [comment]
Please resolve all linting warning on Travis.

**Pull request closed without merging**